### ISSUE TITLE
Upgrading to Rails 4.2: transactional callbacks deprecation warning fires due to the file require order

### ISSUE THREAD

**Issue by Envek** [pat/thinking-sphinx#867]
I'm trying to upgrade my app to upcoming Rails 4.2 (RC 3) and according to official [upgrading manual](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#error-handling-in-transaction-callbacks) I've placed next string to `config/application.rb`:

```
config.active_record.raise_in_transactional_callbacks = true
```

But I still getting next output immediately after `rails server`:

```
DEPRECATION WARNING: Currently, Active Record suppresses errors raised within `after_rollback`/`after_commit` callbacks and only print them to the logs. In the next version, these errors will no longer be suppressed. Instead, the errors will propagate normally just like in other Active Record callbacks.

You can opt into the new behavior and remove this warning by setting:

  config.active_record.raise_in_transactional_callbacks = true

 (called from block in <module:Base> at /Users/anovikov/.rvm/gems/ruby-2.1.5-gost/bundler/gems/thinking-sphinx-43110fdcf9e8/lib/thinking_sphinx/active_record/base.rb:8)
=> Booting WEBrick
=> Rails 4.2.0.rc3 application starting in development on http://0.0.0.0:3000
=> Run `rails server -h` for more startup options
=> Ctrl-C to shutdown server
[2014-12-18 12:44:53] INFO  WEBrick 1.3.1
[2014-12-18 12:44:53] INFO  ruby 2.1.5 (2014-11-13) [x86_64-darwin14.0]
[2014-12-18 12:44:53] INFO  WEBrick::HTTPServer#start: pid=72022 port=3000
```

After some playing with `puts` in `config/application.rb` I've encountered that thinking_sphinx's file [`/lib/thinking_sphinx/active_record/base.rb`](https://github.com/pat/thinking-sphinx/blob/develop/lib/thinking_sphinx/active_record/base.rb) is getting required before the most part of my app's `config/application.rb` is executed.

At this point I've using code from today's `develop` branch of thinking sphinx. I've tried to roll back to published version 3.1.2 and then I've get this warning issued somewhere from `bundler`: `~/.rvm/gems/ruby-2.1.5-gost@global/gems/bundler-1.7.6/lib/bundler/runtime.rb:76`. I've played with puts more and encountered that it's issued while requiring gem [better_errors](https://github.com/charliesome/better_errors), but it seems unrelated to Active Record at all. Weird.

If I remove thinking sphinx from Gemfile at all, warning will disappear.

Any ideas how to solve this?

See https://github.com/rails/rails/issues/18084#issuecomment-67503032 for details.

**Comment by maurosbu** [comment]
Hi @pat,
do you have a schedule for rails 4.2 support? When will this fix be applied to a stable version of TS?
Many thanks,
Mauro

**Comment by pat** [comment]
Just published TS v3.1.3, which includes the Rails 4.2 patches.

**Comment by maurosbu** [comment]
Great!
Thank you @pat!
